src/google/protobuf/testing/googletest.cc

```diff
@@ -94,6 +94,10 @@ string TestSourceDir() {
 namespace {
 
 string GetTemporaryDirectoryName() {
+  char *from_environment = getenv("TEST_TMPDIR");
```

Comment:
Bazel doesn't show an error message currently, but they're moving towards the ability to do that by sandboxing all external commands (not sure if they will catch this specific one or not). It's a "should" in [Bazel's test environment documentation](http://bazel.io/docs/test-encyclopedia.html), so they're leaving it open for now.

I don't think it actually causes any problems if you do it safely (which I believe protobuf does), but it's hard to tell if a use of /tmp is actually safe. After fighting tests (from other projects) which use /tmp unsafely and become flaky when run concurrently, I changed my local version of Bazel's test wrapper so all tests get a read-only /tmp.